homeassistant/components/media_player/cmus.py

```diff
@@ -0,0 +1,223 @@
+"""
+Support for interacting with and controlling the cmus music player.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.mpd/
+"""
+
+import logging
+
+import eventlet
+
+from homeassistant.components.media_player import (
+    MEDIA_TYPE_MUSIC, MEDIA_TYPE_PLAYLIST, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_SET, SUPPORT_PLAY_MEDIA, SUPPORT_SEEK,
+    MediaPlayerDevice)
+from homeassistant.const import (STATE_OFF, STATE_PAUSED, STATE_PLAYING,
+                                 CONF_HOST, CONF_NAME, CONF_PASSWORD,
+                                 CONF_PORT)
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['pycmus>=0.0.3']
+
+SUPPORT_CMUS = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | SUPPORT_TURN_OFF |  \
+    SUPPORT_TURN_ON | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_SEEK
+
+
+def setup_platform(hass, config, add_devices, discover_info=None):
+    """Setup the Cmus platform."""
+    host = config.get(CONF_HOST, None)
+    password = config.get(CONF_PASSWORD, None)
+    port = config.get(CONF_PORT, None)
+    name = config.get(CONF_NAME, None)
+    if host and not password:
+        _LOGGER.error("A password must be set if using a remote cmus server")
+        return False
+    cmus_remote = CmusDevice(host, password, port, name)
+    add_devices([cmus_remote])
+
+
+class CmusDevice(MediaPlayerDevice):
+    """Representation of a running cmus."""
+
+    # pylint: disable=no-member, too-many-public-methods, abstract-method
+    def __init__(self, server, password, port, name):
+        """Initialize the CMUS device."""
+        remote = eventlet.import_patched('pycmus.remote')
```

Comment:
When I was playing with things on an earlier version of the patch (and an earlier version of pycmus) I had some weird socket hanging issues. I wasn't quite sure where they were coming from and wasn't sure where eventlet was being invoked by homeassistant. I've had to workaround similar issues with eventlet and the stdlib socket lib (which is how pycmus works) before. So I wanted to rule that out as a potential cause in debugging those earlier issues by switching the socket import to use the eventlet compatible socket lib instead.

That being said the component code and pycmus have improved a bit since that early test. I'll try switching this back to a regular import and assuming those early issues are resolved I'll remove this.